### PR TITLE
Integration Final Fix - Addresses all Issues

### LINKED ISSUE #170: State Storing Issues
Integration Issues: Storing States

1. Player role not being set on the backend
* Fix GamestartEvent.js - for initial role allocation.
* Line 39  - deep copy player - not the actual player object should do players.find((p) => p.nickname

2. Inconsistent VoteMaps
* Key - nickname, value: player object (day & trial) - to be fixed in Event.js files respectively

3. Player Chosen
* setIsAlive - is never done currently - this needs to be added to endNight and endTrial

4. Medic, sheriff are not included into winning players when civilian wins
* Call getwinningPlayers() instead of get playersbyRole in NightstateEvent.js and TrialStateEvent.js

### CheckList:
- [x] Error 1
- [x] Error 2
- [x] Error 3
- [x] Error 4

### Issue Approval:
- [x]  Issue approved by: @ateagit

### PR DESCRIPTION
**Pull request by MKTPR**
# Description
This PR fixes #170 
All the remaining changes required for integration has been added on to this PR. This involves changing frontend logic and setting player isAlive functionalities. 

# Type of change:
* Bug fix (Non-breaking change which fixes a bug)

# Checklist:
* Have you merged main into your branch?
* Have you tested your changes to ensure it works as expected and does not break existing functionality?
* If applicable, please ensure sufficient tests are added that is related to the changes.
* Please ensure this PR has a `label`, is linked to an `issue` and is related to a `project`
* If new documentation is required for this change, have you created a new documentation issue that describes the documentation needed?

### CODE REVIEW

File: backend/Events/GameStateEvents/NightStateChangeEvents.js [hunk 1]
```diff
@@ -26,7 +24,7 @@ function startNight(io, socket, mafiaGame) {
 
         io.in(roomID).emit('night-start', new NightStartDTO(TIME_TO_VOTE));
 
-        setTimeout(endNight(io, socket, mafiaGame), TIME_TO_VOTE);
+        setTimeout(() => endNight(io, socket, mafiaGame), TIME_TO_VOTE); // TODO fixed bug
```
Comment (SreenikethRaghavan) [inline backend/Events/GameStateEvents/NightStateChangeEvents.js:27]: Is this bug yet to be resolved

File: frontend/src/GameUtils.js [hunk 2]
```diff
@@ -3,17 +3,18 @@ export const nightTimeStatus = {
...
+//     } else if (idx === 0) {
+//         str += p;
+//     } else {
+//         str += `, ${p}`;
+//     }
 
-        return str;
-    });
+//     return str;
+// });
```
Comment (SreenikethRaghavan) [inline frontend/src/GameUtils.js:20]: Can we remove commented out code please if it isn't required?

File: frontend/src/Socket.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import socketIOClient from 'socket.io-client';
 
-const socket = new socketIOClient('mafia-hands-backend-cd.herokuapp.com');
+// const socket = new socketIOClient('mafia-hands-backend-cd.herokuapp.com');
```
Comment (SreenikethRaghavan) [inline frontend/src/Socket.js:3]: Commented out code

**Review by SreenikethRaghavan — COMMENTED** [review]
Just a few questions regarding the changes

**Review by Dhors — APPROVED** [review]
map the ListVoteDTO inside the class itself

File: backend/Events/GameStateEvents/NightStateChangeEvents.js (hunk 1 above)
Comment (Dhors) [reply backend/Events/GameStateEvents/NightStateChangeEvents.js:27]: removed

File: frontend/src/Socket.js (hunk 3 above)
Comment (Dhors) [reply frontend/src/Socket.js:3]: we need this

**Review by Dhors — APPROVED** [review]

**Review by parfei — APPROVED** [review]

**Review by ateagit — APPROVED** [review]

**Pull request merged**